### PR TITLE
Feature/shopping cart

### PR DESCRIPTION
**Pull request by lanavarro-meli**

### CODE REVIEW

File: src/main/java/br/com/dh/meli/projeto_integrador/dto/CustomerDTO.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package br.com.dh.meli.projeto_integrador.dto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.validation.constraints.NotNull;
+
+public class CustomerDTO {
+    @NotNull(message = "Nome do cliente é obrigatório")
+    @JsonProperty("costumer_name")
```
Comment (evycoliveira) [inline src/main/java/br/com/dh/meli/projeto_integrador/dto/CustomerDTO.java:9]: Mudar para@JsonProperty("customer_name").

File: src/main/java/br/com/dh/meli/projeto_integrador/model/Customer.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+@Data
+@Builder
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+public class Customer extends Person{
+    @OneToOne
+    @JoinColumn(name = "shoppingcart_id", nullable = true)
+    private ShoppingCart shoppingCart;
+    @OneToMany(mappedBy = "costumer")
```
Comment (evycoliveira) [inline src/main/java/br/com/dh/meli/projeto_integrador/model/Customer.java]: Alterar para @OneToMany(mappedBy = "customer")

File: src/main/java/br/com/dh/meli/projeto_integrador/model/Customer.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    @JsonIgnoreProperties("costumer")
```
Comment (evycoliveira) [inline src/main/java/br/com/dh/meli/projeto_integrador/model/Customer.java]: Alterar para@JsonIgnoreProperties("customer").

File: src/main/java/br/com/dh/meli/projeto_integrador/model/Order.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private Integer orderNumber;
+    private LocalDateTime orderDate;
+    private Status orderStatus;
+    @OneToMany(mappedBy = "order",fetch = FetchType.LAZY)
+    @JsonIgnoreProperties("order")
+    private List<Item> items;
+    @ManyToOne
+    @JoinColumn(name = "costumer_id", nullable = false)
```
Comment (evycoliveira) [inline src/main/java/br/com/dh/meli/projeto_integrador/model/Order.java]: @JoinColumn(name = "customer_id", nullable = false)

File: src/main/java/br/com/dh/meli/projeto_integrador/model/Order.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    @JsonIgnoreProperties("order")
+    private Customer costumer;
```
Comment (evycoliveira) [inline src/main/java/br/com/dh/meli/projeto_integrador/model/Order.java]: private Customer customer;

File: src/main/java/br/com/dh/meli/projeto_integrador/service/CostumerService.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+import br.com.dh.meli.projeto_integrador.dto.CustomerDTO;
+import br.com.dh.meli.projeto_integrador.exception.NotFoundException;
+import br.com.dh.meli.projeto_integrador.model.Customer;
+import br.com.dh.meli.projeto_integrador.repository.ICustomerRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Optional;
+
+public class CostumerService  implements ICostumerService{
```
Comment (evycoliveira) [inline src/main/java/br/com/dh/meli/projeto_integrador/service/CostumerService.java]: Alterar para public class CustomerService  implements ICustomerService.

File: src/main/java/br/com/dh/meli/projeto_integrador/service/CostumerService.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    @Autowired
+    private ICustomerRepository repo;
+
+    @Override
+    public Customer getCostumerById(Long id) {
```
Comment (evycoliveira) [inline src/main/java/br/com/dh/meli/projeto_integrador/service/CostumerService.java]: Trocar para getCustomerById.

File: src/main/java/br/com/dh/meli/projeto_integrador/service/CostumerService.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+        Optional<Customer> costumer = repo.findById(id);
```
Comment (evycoliveira) [inline src/main/java/br/com/dh/meli/projeto_integrador/service/CostumerService.java]: Optional<Customer> customer = repo.findById(id);

**Review by evycoliveira — APPROVED** [review]

**Review by isaiasfmeli — APPROVED** [review]

**Pull request merged**